### ISSUE TITLE
Incorrect project structure gives an unhelpful error message

### ISSUE THREAD

**Issue by Ekberg** [cucumber/godog#383]
Please answer these questions before submitting a bug report.

### What version of godog are you using?
```
0.11.0
```

### What version of Go are you using?
```
1.16.2
```

### What did you do?
Created a new project with the following structure,
```
|-- features
|   `-- health.feature
|-- main.go
|-- Dockerfile
|-- go.mod
`-- go.sum
```
My `main.go`,
```go
package main

import "github.com/cucumber/godog"

func InitializeScenario(ctx *godog.ScenarioContext) {

}
```
I wanted to confirm that my `Dockerfile` was correctly working and running this minimal example with `godog`. But it failed, I then tried running it with my local copy of `godog`, but that failed as well.

After doing the following change it worked,
```shell
$ mv main.go main_test.go
```

### What did you expect to see?
A more informative error message on why `godog` failed with my minimal example.

### What did you see instead?
```shell
$ godog
failed to compile testmain package: exit status 1 - output: compile: -importcfg: open /tmp/go-build1250231118/b001/importcfg.link: no such file or directory
```

**Comment by mattwynne** [comment]
Yesterday @ALCooper12 and I paired on this problem. We've made a bit of a start on it by adding an example in the `_examples` folder in her fork, where we can reproduce the problem.

I'm new to Go so this is fun!

We're meeting up again next week, when we'll see if we can work out where we'd introduce a change into the code to give a better error message. If anyone has any pointers, we'd love to hear them. I've never worked in this codebase before.
